vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/executor/listeners/video/ScreenRecorderTestBatchListener.kt

```diff
@@ -122,29 +124,56 @@ class ScreenRecorderTestBatchListener(
     }
 
     private suspend fun pullTestVideo(test: Test) {
-        val localVideoFile = fileManager.createFile(FileType.VIDEO, pool, device.toDeviceInfo(), test, testBatchId)
+        val localVideoFiles = mutableListOf<File>()
         val remoteFilePath = device.fileManager.remoteVideoForTest(test, testBatchId)
         val millis = measureTimeMillis {
-            device.safePullFile(remoteFilePath, localVideoFile.toString())
+            if(device.apiLevel >= 34 || videoConfiguration.timeLimit <= 180) {
+                val localVideoFile = fileManager.createFile(FileType.VIDEO, pool, device.toDeviceInfo(), test, testBatchId)
+                device.safePullFile(remoteFilePath, localVideoFile.toString())
+                localVideoFiles.add(localVideoFile)
+            } else {
+                for (i in 0 .. (videoConfiguration.timeLimit / 180)) {
```

Comment:
I suspect that the actual number of files depends on the time the test took, not the configuration limit, so the loop should take into account only the time passed or/and number of shell invocations with `screenrecord` binary